lib/src/components/NxList/__tests__/NxList.test.tsx

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { shallow } from 'enzyme';
+import React from 'react';
+import { NxLoadingSpinner } from '../../..';
+import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
+import NxList from '../NxList';
+import { NxListProps } from '../types';
+
+describe('NxList', function() {
+  const minimalProps: NxListProps = {};
+  const getNxList = enzymeUtils.getShallowComponent<NxListProps>(NxList, minimalProps);
+
+  it('renders a list', function() {
+    const nxList = getNxList();
+    expect(nxList.find('ul')).toMatchSelector('.nx-list');
+  });
+
+  it('renders the classNames given to it', function() {
+    const extendedProps: Partial<NxListProps> = {
+      className: 'test-classname ufo'
+    };
+    const nxList = getNxList(extendedProps);
+    expect(nxList.find('ul')).toMatchSelector('.nx-list.test-classname.ufo');
+  });
+
+  it('passes props to the parent component and renders bulleted list', function() {
+    const contentEl = shallow(
+      <NxList bulleted></NxList>
+    );
+    expect(contentEl.find('ul')).toHaveClassName('nx-list--bulleted');
+  });
+
+  it('renders the children in an .nx-list__item', function() {
+    const children = [
+      <NxList.Item key="1">
+        <NxList.Text>Test Item 1 Text</NxList.Text>
+        <NxList.Subtext>Test Item 1 Subtext</NxList.Subtext>
+      </NxList.Item>,
+      <NxList.Item key="2">
+        <NxList.Text>Test Item 2 Text</NxList.Text>
+        <NxList.Subtext>Test Item 2 Subtext</NxList.Subtext>
+      </NxList.Item>
+    ];
+
+    const contentEl = getNxList({children}).find('ul');
+    expect(contentEl).toExist();
+    expect(contentEl).toContainMatchingElements(2, NxList.Item);
+    contentEl.find('li').forEach((e) => {
+      expect(e.hasClass('nx-list__item'));
+    });
+  });
+
+  it('renders a simple list with a title', function() {
+    const title = 'Simple Title';
+    const children = [
+      <NxList.Title key="1">{title}</NxList.Title>
+    ];
+
+    const contentEl = getNxList({children});
+    expect(contentEl.find('NxListTitle')).toExist();
+    expect(contentEl.find('NxListTitle')).toHaveHTML(`<h3 class="nx-h3">${title}</h3>`);
+  });
+
+  it('renders a clickable list whose list item element has prop disabled and is disabled', function() {
```

Comment:
I'd be more than ok with these subcomponent-oriented tests being split into separate files for each subcomponent, whcih don't even involve the `<NxList>` wrapper but rather just test the subcomponent by itself

Reply:
> I'd be more than ok with these subcomponent-oriented tests being split into separate files for each subcomponent, whcih don't even involve the `<NxList>` wrapper but rather just test the subcomponent by itself

6a99ffed7d912358411bebffe08051b540492dca